qutebrowser/keyinput/eventfilter.py

```diff
@@ -118,8 +120,92 @@ def eventFilter(self, obj: Optional[QObject], event: Optional[QEvent]) -> bool:
             raise
 
 
+class IMEEventHandler(QObject):
+    """Handles the cursorRectangleChanged event from the inputMethod.
+
+    This provides better input detection
+    """
+
+    _last_seen_rect: Optional[QRectF] = None
+
+    def __init__(self, parent: QObject = None) -> None:
+        super().__init__(parent)
+        self._input_method = QApplication.inputMethod()
+        assert self._input_method is not None
+        self._input_method.cursorRectangleChanged.connect(
+            self.cursor_rectangle_changed
+        )
+        self._last_seen_rect = None
+
+    @pyqtSlot()
+    def cursor_rectangle_changed(self) -> None:
+        """Handler for when rectangle changes.
+
+        Event handler for cursorRectangleChanged
+        """
+        # todo:
```

Comment:
What about all those? @toofar posted this as a "very WIP" patch, so I'm wary of merging it in this state without having any rationale of why those things don't matter enough to be fixed first.